src/modules/syslink/syslink_bridge.cpp

```diff
@@ -111,21 +111,45 @@ SyslinkBridge::read(struct file *filp, char *buffer, size_t buflen)
 ssize_t
 SyslinkBridge::write(struct file *filp, const char *buffer, size_t buflen)
 {
-	crtp_message_t msg;
 
-	// Queue and send next time we get a RAW radio packet
-	int remaining = buflen;
+	static bool init = 1;
+
+	if (init) {
+		_msg_to_send.header = 0;
+		_msg_to_send.size = sizeof(_msg_to_send.header);
+		_msg_to_send.port = CRTP_PORT_MAVLINK;
+		init = 0;
+	}
+
+	static int msg_rem = CRTP_MAX_DATA_SIZE -
```

Comment:
Can you move `msg_rem` and `last_index` as class members (and rename to something like `_msg_to_send_size_remaining`). There is a redundancy in msg_rem and last_index: the sum of them is constant so you need to store only one of them.
Generally you should avoid static variables within a method as much as possible.